src/NUnitFramework/framework/Assert.That.cs

```diff
@@ -38,9 +42,31 @@ public partial class Assert
         /// an <see cref="AssertionException"/>.
         /// </summary> 
         /// <param name="condition">The evaluated condition</param>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void That(bool condition, Func<ConstraintResult, string> getExceptionMessage)
+        {
+            Assert.That(condition, Is.True, getExceptionMessage);
+        }
+
+        /// <summary>
+        /// Asserts that a condition is true. If the condition is false the method throws
+        /// an <see cref="AssertionException"/>.
+        /// </summary> 
+        /// <param name="condition">The evaluated condition</param>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void That(bool condition, Func<string> getExceptionMessage)
+        {
+            Assert.That(condition, Is.True, getExceptionMessage);
+        }
+
+        /// <summary>
+        /// Asserts that a condition is true. If the condition is false the method throws
+        /// an <see cref="AssertionException"/>.
+        /// </summary> 
+        /// <param name="condition">The evaluated condition</param>
         /// <param name="message">The message to display if the condition is false</param>
         /// <param name="args">Arguments to be used in formatting the message</param>
-        static public void That(bool condition, string message, params object[] args)
+        public static void That(bool condition, string message, params object[] args)
```

Comment:
Please don't make stylistic changes while making substantive changes. It just confuses the reviewers! At this point, however, let's not change it back but remember for the future.